osu.Game/Beatmaps/ControlPoints/ControlPoint.cs

```diff
@@ -14,6 +14,16 @@ public class ControlPoint : IComparable<ControlPoint>, IEquatable<ControlPoint>
 
         public int CompareTo(ControlPoint other) => Time.CompareTo(other.Time);
 
-        public bool Equals(ControlPoint other) => Time.Equals(other?.Time);
+        /// <summary>
+        /// Whether this <see cref="ControlPoint"/> provides the same changes to gameplay as another <see cref="ControlPoint"/>.
+        /// </summary>
+        /// <param name="other">The <see cref="ControlPoint"/> to compare to.</param>
+        /// <returns>Whether this <see cref="ControlPoint"/> provides the same changes to gameplay as <paramref name="other"/>.</returns>
+        public virtual bool ChangeEquals(ControlPoint other) => !ReferenceEquals(null, other);
+
+        public bool Equals(ControlPoint other)
+            => ChangeEquals(other)
+               && !ReferenceEquals(null, other)
```

Comment:
can you explain why we're using `ReferenceEquals` here as opposed to `Time?.Equals(other?.Time)` or similar?